src/form.spec.ts

```diff
@@ -471,7 +471,7 @@ describe(Form, () => {
         const form = createForm({
           value: { emails: [] },
         });
-        const emails = form.fields.emails;
+        const emails = form.fields.emails as FieldSet<string>;
```

Comment:
(probably that's the reason why you need to export the fieldset type - I personally think this should be avoided, but I didn't give it enough thought yet.

Reply:
Yes I'm aware. This is something i'm struggling with. How can the correct type be inferred if the field is yet undefined?

Should we get rid of the FieldSet subtype and allow any field to potentially have the Fieldset implementation? add, remove, elements?

Reply:
It's just a typing issue, not a runtime issue. That means it only needs to be adressed in the mapped fields interface. Basically what you want is to unwrap the optional for arrays there, so something like `array | undefined` instead of `array`.

Reply:
let me know if you need any help with prototyping those type conditions